build/ExecuteApiCompat.ps1

```diff
@@ -0,0 +1,276 @@
+    
+    # Get Zipfile for ApiCompat
+    $DestinationPath= "$Path\ApiCompat"
+    
+    # If file doesn't exist, download it
+    if (!(Test-Path "$ApiCompatPath\ApiCompat.zip" -PathType Leaf)) {
+        try { 
+            # Download
+            Write-Host "Downloading ApiCompat version $ApiCompatVersion as zip"
+            
+            try {
+                (New-Object System.Net.WebClient).DownloadFile($ApiCompatDownloadRequestUri, $ZipPath)
+            } catch { 
+                Write-Error "Download attempt failed"
+                Write-Error $_
+                exit 1 
+            }
+            
+            $Zip = [ZipFile]::OpenRead($ZipPath)
+            # Extract files
+            try {
+                # Get parent folder from latest apiCompat framework version
+                $ZipApiCompatParentFolder = $zip.Entries | where {$_.Name -like 'Microsoft.DotNet.ApiCompat.exe'} | Select -Last 1
+                $ZipApiCompatParentFolder = split-path -parent $ZipApiCompatParentFolder 
+                $ZipApiCompatParentFolder = split-path -leaf $ZipApiCompatParentFolder
+
+                # Extract all files in latest framework version folder
+                $Zip.Entries.Where{ $_.FullName -match "$ZipApiCompatParentFolder.*[^/]$" }.ForEach{
+                    $NewFile = [IO.FileInfo]($DestinationPath,$_.FullName -join "/")
+                    $NewFile.Directory.Create()
+                    [ZipFileExtensions]::ExtractToFile($_, $NewFile)
+                }
+            } catch {
+                Write-Error "Failed to extract files from $ZipPath"
+                exit 6
+            } finally {
+                $Zip.Dispose()
+            }
+
+            Write-Host "$ZipFile successfully downloaded and extracted" -ForegroundColor green
+        } finally {
+            # Remove downloaded zip file.
+            Remove-Item $ZipFile
+        }
+    }
+
+    cd $Path
+}
+
+$WriteToLog = {
+    $ResultMessage = " Binary Compatibility compared against version: $Version => $ApiCompatResult"
+    Write-Host $ResultMessage -ForegroundColor green
+    
+    # Create a Mutex for all process to be able to share same log file
+    $mutexName = "LogFileMutex" #'A unique name shared/used across all processes that need to write to the log file.'
+    $mutex = New-Object 'Threading.Mutex' $false, $mutexName
+    
+    #Grab the mutex. Will block until this process has it.
+    $mutex.WaitOne() | Out-Null;
+    
+    try {
+        # Now it is safe to write to log file
+        Add-Content $OutputDirectory $ResultMessage
+    } finally {
+        $mutex.ReleaseMutex()
+    }
+}
+
+# Get specific dll file from built solution
+$Dll = Get-ChildItem "$Path\**\$Name\bin\Debug\**\**" -Filter "$Name.dll" | % { $_.FullName } | Select -First 1
+if ([string]::IsNullOrEmpty($Dll)){
+    $Dll = Get-ChildItem "$Path\**\**\$Name\bin\Debug\**\**" -Filter "$Name.dll" | % { $_.FullName } | Select -First 1
+
+    if ([string]::IsNullOrEmpty($Dll)) {
```

Comment:
Are we getting the DLLs from Debug only? If this is the case, we should add this information to the readme.

Reply:
Instead of limiting it only to the Debug configuration, I added the ability to choose whether to compare local Release or Debug dlls by adding the corresponding parameters.